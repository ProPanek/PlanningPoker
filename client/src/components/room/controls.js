import React, {Component} from 'react'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Card";
import styled from "styled-components";
import {Button, Card} from "@material-ui/core";
import {decorate, observable} from "mobx";

const StyledButtonCard = styled(Button)`
  &&{
  width: 10%;
  height: 5vh;
  margin:5px;
  flex:1;
  }
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 83.333333%;
  height: 10vh;
  flex-wrap: wrap;
`;

const Wrapper = styled(Card)`
  &&{
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 90%;
  flex-wrap: wrap;
  margin-top:10px;
  margin-bottom: 10px;
  padding: 5px;
  justify-content: center;
  }
`;
const cards = [0.5, 1, 2, 3, 5, 8, 13, 20, 40, 100]

class Controls extends Component {


  handleReset = () => {
    this.props.store.resetCards()
    this.props.store.cardResults = []
  }

  handleCard = () => {
    this.props.store.sendCard(this.selectedCard)
    this.props.store.blockCard = true
  }

  chooseCard = (card) => {
    this.selectedCard = card
  }

  render() {
    return (
      <React.Fragment>
        <Wrapper> We're waiting for : {this.props.store.waiting} users </Wrapper>
        <Button disabled={this.props.store.blockCard} variant="contained" color="secondary"
                onClick={this.handleCard}>Send Card</Button>
        {this.props.store.admin && (
          <Button onClick={this.handleReset}>Next issue</Button>
        )}
        <ButtonWrapper>
          {cards.map((card, index) => {
            return (
              <StyledButtonCard variant="contained" color="primary" key={index} onClick={() => {
                this.chooseCard(card)
              }}>{card}</StyledButtonCard>
            )
          })}
        </ButtonWrapper>
      </React.Fragment>
    )
  }
}

decorate(Controls, {
  selectedCard: observable,
});


export default inject("store")(withRouter(observer(Controls)));

