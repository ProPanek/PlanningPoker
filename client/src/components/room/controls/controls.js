import React, {Component} from "react"
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import styled from "styled-components";
import Card from "@material-ui/core/Card";
import Button from "@material-ui/core/Button";
import {decorate, observable} from "mobx";

const StyledButtonCard = styled(Button)`
  &&{
  width: 10%;
  height: 10vh;
  margin:5px;
  flex:1;
  }
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 83.333333%;
  height: 10vh;
  flex-wrap: wrap;
`;

const YourCard = styled(Card)`
  margin: 0 auto;
  margin-top:10px;
  width: 100px;
  height: 100px;
  font-size:50px;
  line-height: 100px;
  text-align: center;
`;

const cards = ["â˜•",0,0.5, 1, 2, 3, 5, 8, 13, 20, 40, 100]

class Controls extends Component {
  constructor(props) {
    super(props)
    this.selectedCard = ""
  }

  handleReset = () => {
    this.props.store.resetCards()
    this.props.store.room.cardResults = []
    this.props.store.jira.issueId = ""
  }

  handleCard = () => {
    this.props.store.sendCard(this.selectedCard)
    this.props.store.blockCard = true
    this.selectedCard = ""
  }

  chooseCard = (card) => {
    this.selectedCard = card
  }

  handleEstimation = () => {
    this.props.store.setIssueEstimation()
    this.props.store.jira.issueId = ""
    this.props.store.selectBoard(this.props.store.jira.boardId)
  }

  render() {
    return (
      <React.Fragment>
        <Button disabled={this.props.store.blockCard} variant="contained" color="secondary"
                onClick={this.handleCard}>Send Card</Button>
        {this.props.store.user.admin && (
          <React.Fragment>
            <Button
              disabled={this.props.store.blockReset}
              onClick={this.handleReset}>Reset cards</Button>
            <Button
              disabled={!this.props.store.room.cardsAreTheSame}
              onClick={this.handleEstimation}>Set estimation point</Button>
          </React.Fragment>
        )}
        {this.selectedCard !== "" && <YourCard> {this.selectedCard}</YourCard>}
        <ButtonWrapper>
          {cards.map((card, index) => {
            return (
              <StyledButtonCard variant="contained" color="primary" key={index} onClick={() => {
                this.chooseCard(card, index);
              }}>{card}</StyledButtonCard>
            )
          })}
        </ButtonWrapper>
      </React.Fragment>
    )
  }
}

decorate(Controls, {
  selectedCard: observable,
});

export {Controls}
export default inject("store")(withRouter(observer(Controls)));

