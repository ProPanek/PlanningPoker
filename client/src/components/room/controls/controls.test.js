import React from "react"
import { Controls } from "./controls"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("Controls", () => {
  it("renders without errors", () => {
    const store = {
      room: { cardsAreTheSame : false, cardResults : []},
      user: {admin: true}
    }

    const component = shallow(<Controls store={store} />)
    expect(component).toMatchSnapshot()
  })
})
