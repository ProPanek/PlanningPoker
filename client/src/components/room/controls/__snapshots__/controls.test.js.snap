// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders without errors 1`] = `
<Fragment>
  <WithStyles(Button)
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    Send Card
  </WithStyles(Button)>
  <WithStyles(Button)
    onClick={[Function]}
  >
    Reset cards
  </WithStyles(Button)>
  <WithStyles(Button)
    disabled={true}
  >
    Set estimation point
  </WithStyles(Button)>
  <styled.div>
    <Styled(WithStyles(Button))
      color="primary"
      key="0"
      onClick={[Function]}
      variant="contained"
    >
      â˜•
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="1"
      onClick={[Function]}
      variant="contained"
    >
      0
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="2"
      onClick={[Function]}
      variant="contained"
    >
      0.5
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="3"
      onClick={[Function]}
      variant="contained"
    >
      1
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="4"
      onClick={[Function]}
      variant="contained"
    >
      2
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="5"
      onClick={[Function]}
      variant="contained"
    >
      3
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="6"
      onClick={[Function]}
      variant="contained"
    >
      5
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="7"
      onClick={[Function]}
      variant="contained"
    >
      8
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="8"
      onClick={[Function]}
      variant="contained"
    >
      13
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="9"
      onClick={[Function]}
      variant="contained"
    >
      20
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="10"
      onClick={[Function]}
      variant="contained"
    >
      40
    </Styled(WithStyles(Button))>
    <Styled(WithStyles(Button))
      color="primary"
      key="11"
      onClick={[Function]}
      variant="contained"
    >
      100
    </Styled(WithStyles(Button))>
  </styled.div>
</Fragment>
`;
